### PR TITLE
Migrate loader settings to use the new API

### PR DESCRIPTION
**Pull request by beraldoleal**
This will migrate `--loaders`, `--external-runner`, `--external-runner-chdir` and `--external-runner-testdir` to use the new API calls.

### CODE REVIEW

File: avocado/core/loader.py [hunk 1]
```diff
@@ -139,8 +139,6 @@ def _str_loaders():
         loaders = args.get('loaders', None)
         if not loaders:
             loaders = settings.get_value("plugins", "loaders", list, [])
-        if '?' in loaders:
```
Comment (clebergnu) [inline avocado/core/loader.py:142]: Oh yeah! But please also remove the mentions from `guides/user/chapters/loaders.rst`

File: avocado/core/loader.py [hunk 2]
```diff
@@ -135,10 +137,10 @@ def _str_loaders():
         supported_types = []
         for plugin in self.registered_plugins:
             supported_types.extend(_good_test_types(plugin))
-        # Load plugin by the priority from settings
-        loaders = args.get('loaders', None)
-        if not loaders:
-            loaders = settings.get_value("plugins", "loaders", list, [])
+
+        # Here is one of the few exceptions that can has a hardcoded default
```
Comment (clebergnu) [inline avocado/core/loader.py:142]: I think you meant: "Here is one of the few exceptions that has a hardcoded default" ?

File: avocado/core/loader.py [hunk 3]
```diff
@@ -411,15 +412,17 @@ def add_loader_options(parser, section='run'):
...
-                        help=('Path to an specific test runner that '
-                              'allows the use of its own tests. This '
-                              'should be used for running tests that '
-                              'do not conform to Avocado\' SIMPLE test'
-                              'interface and can not run standalone. Note: '
-                              'the use of --external-runner overwrites the --'
-                              'loaders to "external_runner"'))
+    help_msg = ("Path to an specific test runner that allows the use of its "
+                "own tests. This should be used for running tests that do not "
+                "conform to Avocado\' SIMPLE test interface and can not run "
```
Comment (clebergnu) [inline avocado/core/loader.py:417]: It was not introduced by your commit, but this should read: "conform to Avocado's SIMPLE test..."

File: avocado/plugins/replay.py [hunk 4]
```diff
@@ -234,7 +234,9 @@ def run(self, config):
                               'ignore_missing_references',
                               'execution_order',
                               'loaders',
-                              'external_runner']:
+                              'external_runner',
+                              'external_runner_testdir',
```
Comment (clebergnu) [inline avocado/plugins/replay.py:238]: It's a small issue, but this line should be in the commit "migrate --external-runner-testdir to use the new settings API" commit (or if you want merge the commits, it's also fine).

**Comment by beraldoleal** [comment]
Hi @clebergnu thanks for your comments, I'm closing this in favor of a v2.

**Pull request closed without merging**